### ISSUE TITLE
Not mount /mnt/boot/efi if have boot on external partition

### ISSUE THREAD

**Issue by bilkoua** [Chrysostomus/aif-dev#228]
for next layout: 

```
NAME           MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINT
sda              8:0    0 931.5G  0 disk  
 ├─sda1           8:1    0   512M  0 part  /mnt/boot/efi
 ├─sda2           8:2    0   200M  0 part  
 │ └─cryptboot  254:0    0   198M  0 crypt /mnt/boot
 ├─sda3           8:3    0   800G  0 part  
 │ └─cryptlvm   254:1    0   800G  0 crypt 
 │   ├─vg0-swap 254:2    0    4G  0 lvm   [SWAP]
 │   ├─vg0-root 254:3    0   20G  0 lvm   /mnt
 │   └─vg0-home 254:4    0   776G  0 lvm   /mnt
```
we have missing 

```
mkdir /mnt/boot/efi
mount /dev/sda1 /mnt/boot/efi
```

and need do it manually

**Comment by Chrysostomus** [comment]
I don't understand. What is the problem? Did you mount /boot at esp mounting phase?

**Comment by bilkoua** [comment]
No, I'll try mounting sda1 for /mnt/boot/efi at esp because /boot (sda2) need encrypting for full encrypt disk

**Comment by Chrysostomus** [comment]
Okay, bug confirmed and replicated.
There are two issues at play here: 

1) separate encrypted /boot is not yet handled by the installer (missing feature) 
2) esp is unmounted when you mount separate /boot. This is probably a natural result of mounting a higher mountpoint in hierarchy. This could be handled by automatically remounting esp to the chosen mountpoint, or by moving the esp mounting to the end of the mount  dialog.

Btw, in case, I would suggest putting that /boot inside the lvm instead. Simpler that way.

**Comment by bilkoua** [comment]
yeap, I was fix it with /boot inside the lvm. For grub2 don't need using /boot outside lvm...

**Comment by Chrysostomus** [comment]
Should be fixed with c1a93e5997ff53fb8da201fe9bd4ce2746da79b6
